matrix/Vector3.hpp

```diff
@@ -83,6 +83,24 @@ class Vector3 : public Vector<Type, 3>
 
 typedef Vector3<float> Vector3f;
 
+template <typename Type>
+class Quaternion;
+
+template<typename Type>
+Vector3<Type> operator* (const Quaternion<Type> &q, const Vector3<Type>& v) {
```

Comment:
I have changed that to be quaternion multiplication formula. I don't know if I previously got "wrong" results because of non hamilton representation of quaternion. I did not have chance to test the nvidia version with new updates yet. It was not completely wrong - just wrong in the context of this library. But for the sake of being really sure that I always get correct result I have changed it to be q(0, vnew) = q * q(0, v) * q.inversed();.